public/js/components/Scopes.js

```diff
@@ -0,0 +1,107 @@
+"use strict";
+
+const React = require("react");
+const { bindActionCreators } = require("redux");
+const { connect } = require("react-redux");
+const actions = require("../actions");
+const { getPause } = require("../selectors");
+const { DOM: dom } = React;
+
+require("./Scopes.css");
+
+function getScopes(pauseInfo) {
+  if (!pauseInfo) {
+    return [];
+  }
+
+  const environment = pauseInfo.getIn(["frame", "environment"]);
+  let scope = environment;
+  const scopes = [ scope ];
+
+  while (scope = scope.get("parent")) { // eslint-disable-line no-cond-assign
+    scopes.push(scope);
+  }
+
+  return scopes;
+}
+
+function getScopeVariables(scope) {
+  return scope.getIn(["bindings", "variables"]);
+}
+
+function renderArgumentProperty(argumentBinding) {
+  const argumentName = argumentBinding.keySeq().first();
+  const actor = argumentBinding.first().getIn(["value", "actor"]);
+  return dom.li({className: "scope-property-list-item", key: actor},
```

Comment:
Nit: we've been discouraging this system in the devtools codebase. We prefer:

```
dom.li(
  { className: "scope", ... },
  ...
)
```

Generally it's easier to add more properties and looks better when there are more. Also some editors (like mine) want to align the arguments. Also, space after and before curly braces.